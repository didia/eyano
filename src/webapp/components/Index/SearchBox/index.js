// Vendor
import React from 'react';
import { injectIntl, intlShape } from 'react-intl';
import SVG from 'react-inlinesvg';

// Assets
import searchIcon from '../../../static/images/icon-magnifying-glass.svg';
import resetIcon from '../../../static/images/icon-close.svg';

class SearchBox extends React.Component {
  constructor() {
    super();

    this.state = {
      searchText: ''
    };

    this.handleSearchTextChange = this.handleSearchTextChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleReset = this.handleReset.bind(this);
  }

  handleReset(event) {
    this.setState({
      ...this.state,
      searchText: ''
    });

    event.preventDefault();
  }

  handleSearchTextChange(event) {
    this.setState({
      ...this.state,
      searchText: event.target.value
    });
  }

  handleSubmit(event) {
    window.console.log(`Search query is: ${this.state.searchText}`);
    event.preventDefault();
  }

  hasInput() {
    return this.state.searchText.trim() !== '';
  }

  computeFormClassName() {
    const formBaseClass = 'form';
    return this.hasInput() ? `${formBaseClass} form--dirty` : formBaseClass;
  }

  render() {
    const placeHolderText = this.props.intl.formatMessage({
      id: 'components.search-box.placeholder'
    });

    return (
      <form
        className={this.computeFormClassName()}
        onSubmit={this.handleSubmit}
        onReset={this.handleReset}
      >
        <div className="input-wrapper">
          <input
            autoFocus={true}
            className="input"
            placeholder={placeHolderText}
            value={this.state.searchText}
            onChange={this.handleSearchTextChange}
          />

          <button className="reset-button" type="reset">
            <SVG
              className="button-wrapper reset-icon-wrapper"
              src={resetIcon}
              wrapper={React.createFactory('div')}
            />
          </button>
        </div>

        <button
          className="search-button"
          type="submit"
          disabled={!this.hasInput()}
        >
          <SVG
            className="button-wrapper search-icon-wrapper"
            src={searchIcon}
            wrapper={React.createFactory('div')}
          />
        </button>

        <style jsx>{`
          .form {
            display: flex;
            align-items: center;
            box-shadow: 0 16px 32px rgba(0, 44, 80, 0.2);;
          }

          .form :global(.button-wrapper) {
            display: flex;
            align-items: center;
          }

          .input-wrapper {
            display: flex;
            align-items: center;
            width: 100%;
            background-color: #fff;
            padding: 16px;
          }

          .input {
            width: 100%;
            outline: none;
            border: none;
            font-size: 20px
            color: #595959;
            -webkit-font-smoothing: antialiased
          }

          .reset-button {
            overflow: hidden;
            margin-right: 4px;
            opacity: 0;
            transition: 0.3s opacity ease-in-out;
            cursor: pointer;
          }

          .reset-button :global(svg) {
            width: 15px;
            height: 15px;
            fill: #555c5c;
          }

          .search-button {
            height: 57px;
            width: 57px;
            flex-shrink: 0;
            cursor: pointer;
            background-color: #006b6b;
            transition: 0.3s background-color ease-in-out;
          }

          .search-button :global(svg) {
            width: 20px;
            height: 20px;
            fill: #fff;
          }

          .search-button:hover,
          .search-button:active {
            background-color: #4cc7c7;
          }

          .search-button:disabled {
            cursor: default;
            background-color: #008989;
          }

          .search-button :global(.search-icon-wrapper) {
            justify-content: center;
          }

          .form--dirty .reset-button {
            opacity: 1;
          }
        `}</style>
      </form>
    );
  }
}

SearchBox.propTypes = {
  intl: intlShape.isRequired
};

export default injectIntl(SearchBox);
