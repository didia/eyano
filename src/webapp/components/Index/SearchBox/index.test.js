// Vendor
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { IntlProvider } from 'react-intl';

import messages from '../../../locales/fr';

import SearchBox from './index';

describe('SearchBox Component', () => {
  it('renders as expected', () => {
    const intlProvider = new IntlProvider({ locale: 'fr', messages }, messages);
    const { intl } = intlProvider.getChildContext();
    const renderer = new ShallowRenderer();

    const component = renderer.render(
      <SearchBox.WrappedComponent intl={intl} />
    );

    expect(component).toMatchSnapshot();
  });
});
