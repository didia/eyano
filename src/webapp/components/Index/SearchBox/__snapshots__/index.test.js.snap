// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component renders as expected 1`] = `
<form
  className="jsx-547727129 form"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="jsx-547727129 input-wrapper"
  >
    <input
      autoFocus={true}
      className="jsx-547727129 input"
      onChange={[Function]}
      placeholder="Trouver la réponse à votre question"
      value=""
    />
    <button
      className="jsx-547727129 reset-button"
      type="reset"
    >
      <InlineSVG
        cacheGetRequests={false}
        className="button-wrapper reset-icon-wrapper"
        onLoad={[Function]}
        src="/static/images/icon-close.svg"
        supportTest={[Function]}
        uniquifyIDs={true}
        wrapper={[Function]}
      />
    </button>
  </div>
  <button
    className="jsx-547727129 search-button"
    disabled={true}
    type="submit"
  >
    <InlineSVG
      cacheGetRequests={false}
      className="button-wrapper search-icon-wrapper"
      onLoad={[Function]}
      src="/static/images/icon-magnifying-glass.svg"
      supportTest={[Function]}
      uniquifyIDs={true}
      wrapper={[Function]}
    />
  </button>
  <JSXStyle
    css=".form.jsx-547727129{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;box-shadow:0 16px 32px rgba(0,44,80,0.2);}.form.jsx-547727129 .button-wrapper{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.input-wrapper.jsx-547727129{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;width:100%;background-color:#fff;padding:16px;}.input.jsx-547727129{width:100%;outline:none;border:none;font-size:20px;color:#595959;-webkit-font-smoothing:antialiased;}.reset-button.jsx-547727129{overflow:hidden;margin-right:4px;opacity:0;-webkit-transition:0.3s opacity ease-in-out;transition:0.3s opacity ease-in-out;cursor:pointer;}.reset-button.jsx-547727129 svg{width:15px;height:15px;fill:#555c5c;}.search-button.jsx-547727129{height:57px;width:57px;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;cursor:pointer;background-color:#006b6b;-webkit-transition:0.3s background-color ease-in-out;transition:0.3s background-color ease-in-out;}.search-button.jsx-547727129 svg{width:20px;height:20px;fill:#fff;}.search-button.jsx-547727129:hover,.search-button.jsx-547727129:active{background-color:#4cc7c7;}.search-button.jsx-547727129:disabled{cursor:default;background-color:#008989;}.search-button.jsx-547727129 .search-icon-wrapper{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}.form--dirty.jsx-547727129 .reset-button.jsx-547727129{opacity:1;}"
    styleId="547727129"
  />
</form>
`;
